libs/shared/ui/src/lib/components/inputs/input-radio/input-radio.tsx

```diff
@@ -1,28 +1,48 @@
-import { useRef } from 'react'
-import './input-radio.module.scss'
+import { useEffect, useState } from 'react'
 
-/* eslint-disable-next-line */
 export interface InputRadioProps {
   name: string,
+  label?: string,
+  value: string,
   isChecked?: boolean,
   className?: string
+  getValue?: (checked: boolean, value: string) => void
 }
 
 export function InputRadio(props: InputRadioProps) {
 
   const {
     name,
+    value,
+    getValue,
     isChecked = false,
-    className = ''
+    className = '',
+    label = '',
   } = props
 
+  const [check, setCheck] = useState(isChecked);
+
+  useEffect(() => {
+    setCheck(isChecked)
+  }, [isChecked])
+
+  const inputChange = (check: boolean, value: string) => {
+    setCheck(check)
+    if (getValue) getValue(check, value)
+  }
+
   return (
-    <div className={className}>
-      <input 
+    <div className={`input__radio flex gap-2 items-center ${className}`}>
+      <input
+        id={value}
         type="radio"
         name={name}
-        checked={isChecked}
+        value={value}
+        checked={check}
+        onChange={(e) => inputChange(e.currentTarget.checked, e.currentTarget.value)}
+        className="relative font-icons w-0 h-0 mr-5 before:absolute before:flex before:justify-center before:items-center before:text-white before:w-5 before:h-5 before:top-0 before:left-0 before:-translate-y-1/2 before:rounded-full before:bg-white before:border-element-light-lighter-600 before:border before:font-black before:text-xs before:leading-none before:content-[''] before:font-icons checked:before:content-['\f00c'] checked:before:bg-brand-500 checked:before:border-brand-500 before:transition-all"
```

Comment:
Two times you set font-icons the `before` must take into account that of the parent